src/ErrorHandler/BlacklistingDebugDataCreator.php

```diff
@@ -0,0 +1,82 @@
+<?php
+namespace ShoppinPal\YapepCommon\ErrorHandler;
+
+use YapepBase\ErrorHandler\DebugDataCreator;
+
+class BlacklistingDebugDataCreator extends DebugDataCreator
+{
+    const BLACKLISTED_CLASS_PATTERNS = [
+        '/(^|\\\\)Aws(\\\\|$)/i',
+        '/(^|\\\\)Amazon(\\\\|$)/i',
+        '/(^|\\\\)Sqs/i',
+        '/(^|\\\\)S3/i',
+        '/(^|\\\\)Config(\\\\|$)/i',
+    ];
+
+    const MAX_DEPTH = 10;
+
+    /**
+     * @param string $errorId
+     * @param string $errorMessage
+     * @param array  $backTrace
+     * @param array  $context
+     *
+     * @return string
+     */
+    protected function getDebugData($errorId, $errorMessage, array $backTrace = array(), array $context = array())
+    {
+        $context = $this->removeBlackListedClassesFromArray($context);
+        $backTrace = $this->removeBlackListedClassesFromArray($backTrace);
+
+        return parent::getDebugData(
+            $errorId,
+            $errorMessage,
+            $backTrace,
+            $context
+        );
+    }
+
+    /**
+     * @param array|\Iterator|\IteratorAggregate $data
+     * @param int                                $recursionDepth
+     *
+     * @return array
+     */
+    protected function removeBlackListedClassesFromArray($data, $recursionDepth = 0)
```

Comment:
With the recursion I agree, and the depth you gave is quite generous, but I don't get why you want to blacklist certain classes

Reply:
Because the AWS classes cause our trace files to be about 50MB big, without it it's about 25KB. The 50MB big trace file is horrible to use. Blacklisting the Config is because of the same reasons as we have for cleaning env values from the $_SERVER array, namely we have secrets in the trace files